### PR TITLE
Fix DeepSeek V4 MTP cudagraph on MUSA

### PR DESCRIPTION
**Pull request by Joey-gvwal**
## Summary

Fix DeepSeek-V4-Flash-Base startup with MTP and FULL_DECODE_ONLY cudagraph on MUSA for the vLLM 0.24.0 adaptation.

## Changes:
- Replaces the old DeepSeek-V4 MTP patch with focused v0.24.0 patch-series entries.
- Keeps `fused_mtp_input_rmsnorm` branch dtypes stable to avoid Triton MUSA compilation failure.
- Skips DeepSeek-V4 sparse MLA mixed warmup for MUSA MTP, which is not compatible with this path.
- Avoids MUSA TP logits gather/all-gather failures by reconstructing logits through TP all-reduce.
- Bypasses `torch.ops._C.weak_ref_tensor` on MUSA to allow decode cudagraph capture.

## Test

```
 export VLLM_WORKER_MULTIPROC_METHOD=spawn
  export PYTHONUNBUFFERED=1
  export VLLM_MUSA_DEEPSEEK_V4_TP8_PROFILE=aggressive_long_prefill
  export VLLM_MUSA_DEEPSEEK_V4_DISABLE_AUX_OVERLAP=1
  export VLLM_MUSA_DEEPSEEK_V4_MHC_PRE_DECODE_PRENORM_IMPL=deepgemm
  export VLLM_MUSA_SPEC_METADATA_SINGLE_REQ_CACHE=1

  unset TORCHDYNAMO_DISABLE
  unset VLLM_ATTENTION_BACKEND

  vllm serve /home/dist/models/DeepSeek-V4-Flash-Base \
    --served-model-name deepseek \
    --gpu-memory-utilization 0.95 \
    --max-model-len 6144 \
    --max-num-seqs 1 \
    --max-num-batched-tokens 8195 \
    --tensor-parallel-size 4 \
    -ac.backend FLASHMLA \
    --port 25606 \
    --host 0.0.0.0 \
    --kv-cache-dtype fp8 \
    --no-enable-prefix-caching \
    --async-scheduling \
    --compilation-config '{"mode":"NONE","cudagraph_mode":"FULL_DECODE_ONLY","max_cudagraph_capture_size":5,"cudagraph_capture_sizes":[5],"cudagraph_copy_inputs":false}' \
    --speculative-config '{"method":"mtp","num_speculative_tokens":4}'
```

Output correctness checks:
- `/v1/completions`: `Question: What is 2+2?` returned `4`.
- `/v1/completions`: `The capital of France is` returned `Paris`.
- Both requests returned HTTP 200 and stopped without special-token leakage.

### CODE REVIEW

File: vllm_musa/patches/series/0060-MUSA-vllm.models.deepseek_v4.nvidia.mtp.patch [hunk 1]
```diff
@@ -1,25 +0,0 @@
-From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
```
[the remark this answers is not in the document]
Comment (Joey-gvwal) [reply vllm_musa/patches/series/0060-MUSA-vllm.models.deepseek_v4.nvidia.mtp.patch:1]: No fix is needed for this commit, 0060-MUSA-vllm.models.deepseek_v4.nvidia.mtp.patch has been deleted.
